### PR TITLE
[IDEA-299573] Handle Optional.isEmpty() in "Non functional style Optional.isPresent() usage" inspection

### PR DESCRIPTION
**Pull request by Fabrice-TIERCELIN**
Hi @BasLeijdekkers,

This PR handles `Optional.isEmpty()` as it handles `Optional.isPresent()`. It wasn't the case before because the inspection has been created in 2016 and `Optional.isEmpty()` appeared in Java 11 (2018). This PR also handles multiple negations:

```Java
  public void testOptional(java.util.Optional<String> str) {
    String val;
    if (!!!!!!!!!!!!!!!!str.isEmpty()) {
      val = "";
    } else {
      val = str.get();
    }
  }
```

This PR also adds some unit tests.

### CODE REVIEW

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// "Replace Optional.isPresent() condition with functional style expression" "true-preview"
```
Comment (amaembo) [inline java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java:1]: I see unrelated changes. E. g., removal of `-preview` here. Please rebase your patch on top of master properly to avoid unrelated changes. Thanks.

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+// "Replace Optional presence condition with functional style expression ''Optional.ifPresent()''" "true"
```
Comment (amaembo) [inline java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java:1]: Double single quotes should not be used in messages.

File: java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java [hunk 3]
```diff
@@ -131,10 +124,20 @@ private static PsiStatement extractElseStatement(@NotNull PsiIfStatement ifState
   }
 
   @Nullable
-  @Contract("null -> null")
-  static PsiReferenceExpression extractOptionalFromIsPresentCheck(PsiExpression expression) {
-    PsiMethodCallExpression call = ObjectUtils.tryCast(expression, PsiMethodCallExpression.class);
-    if (!OPTIONAL_IS_PRESENT.matches(call)) return null;
+  @Contract("null, _ -> null")
+  private static PsiReferenceExpression extractOptionalFromIsPresentCheck(PsiExpression condition, boolean[] invert) {
```
Comment (amaembo) [inline java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java:128]: Returning value in parameter is not a good practice in Java. Especially, using one-element array for this purpose. Is it possible to avoid such a pattern?

File: java/openapi/resources/messages/JavaBundle.properties [hunk 4]
```diff
@@ -101,9 +102,7 @@ checkbox.param.indent.on.continuation=Indent continuation lines
 checkbox.prefer.longer.names=Prefer longer names
 checkbox.preserve.line.feeds=Preserve line feeds
 checkbox.spaces.record.header=Record header
-checkbox.spaces.within.deconstruction.list=Deconstruction list
 checkbox.spaces.before.colon.in.foreach=Before colon in foreach
-checkbox.spaces.before.deconstruction.list=Deconstruction list
```
Comment (amaembo) [inline java/openapi/resources/messages/JavaBundle.properties:106]: Unrelated change, please rollback

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAssignmentField.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-// "Replace Optional.isPresent() condition with functional style expression" "true-preview"
+// "Replace Optional presence condition with functional style expression ''Optional.ifPresent()''" "true"
```
Comment (amaembo) [inline java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAssignmentField.java:1]: The fix name is clearly incorrect now: it says that "Optional.ifPresent" will be used while in fact, `orElse` is used.

**Review by amaembo — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please address the problems in comments.

**Comment by Fabrice-TIERCELIN** [comment]
Sorry, incorrect use

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java (hunk 1 above)
Comment (Fabrice-TIERCELIN) [reply java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java:1]: Fixed

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java (hunk 2 above)
Comment (Fabrice-TIERCELIN) [reply java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAnnotation.java]: Fixed

File: java/openapi/resources/messages/JavaBundle.properties (hunk 4 above)
Comment (Fabrice-TIERCELIN) [reply java/openapi/resources/messages/JavaBundle.properties]: Rollbacked

File: java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAssignmentField.java (hunk 5 above)
Comment (Fabrice-TIERCELIN) [reply java/java-tests/testData/codeInsight/daemonCodeAnalyzer/quickFix/optionalIsPresent/afterAssignmentField.java]: Removed

File: java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java (hunk 3 above)
Comment (Fabrice-TIERCELIN) [reply java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java:128]: It would be far more complex without it. Do you prefer an `AtomicBoolean` ?

Comment (amaembo) [reply java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java:128]: Of course, not. Ok, will fix it by myself.

**Comment by amaembo** [comment]
1c9de6b4cf43c2035467b97228e28f15fd3c4879
